lib/test_formulae.rb

```diff
@@ -15,22 +15,25 @@ def initialize(tap:, git:, dry_run:, fail_fast:, verbose:)
 
       protected
 
+      def cached_event_json
+        return unless (event_json = artifact_cache/"event.json").exist?
+
+        event_json
+      end
+
       def previous_github_sha
         return if tap.blank?
         return unless repository.directory?
         return if ENV["GITHUB_ACTIONS"].blank?
+        return if (github_event_path = ENV.fetch("GITHUB_EVENT_PATH", nil)).blank?
 
-        @previous_sha ||= begin
-          event_path = ENV.fetch("GITHUB_EVENT_PATH")
-          event_payload = JSON.parse(File.read(event_path))
+        github_event_payload = JSON.parse(File.read(github_event_path))
+        return if github_event_payload.dig("pull_request", "head", "repo", "owner", "login") != "Homebrew"
```

Comment:
```suggestion
        return if github_event_payload.dig("pull_request", "head", "repo", "owner", "login") != ENV.fetch("GITHUB_REPOSITORY_OWNER")
```

Maybe?

Will allow it to work in third-party taps.

Reply:
Thanks, applied.